sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubIO.java

```diff
@@ -524,6 +524,16 @@ public static Read<PubsubMessage> readMessagesWithAttributesAndMessageId() {
         .build();
   }
 
+  /**
+   * Returns A {@link PTransform} that continuously reads from a Google Cloud Pub/Sub stream. The
+   * messages will contain a {@link PubsubMessage#getPayload() payload}, {@link
+   * PubsubMessage#getAttributeMap() attributes}, along with the {@link PubsubMessage#getMessageId()
+   * messageId} and {PubsubMessage#getOrderingKey() orderingKey} from PubSub.
+   */
+  public static Read<PubsubMessage> readMessagesWithAllAttributesAndMessageIdAndOrderingKey() {
```

Comment:
Can we remove "all" from the message name to keep it consistent with the above method?
Also, based on the above method, do we need .setNeedsAttributes(true) here?
Finally, should we name the coder PubsubMessageCoder, given that the above method calls it PubsubMessageWithAttributesAndMessageIdCoder?

Reply:
I think that since this coder grabs all fields of an incoming PubsubMessage then it could be named without specifying all these fields in its name. Accordingly, method would be better named 'readMessages` but this name is already taken by other implementation (that reads only payload of PubsubMessage).

I agree that `readMessagesWithAllAttributesAndMessageIdAndOrderingKey` then should be named `readMessagesWithAttributesAndMessageIdAndOrderingKey`, so I'm updating it to this.

Regardint the
> Also, based on the above method, do we need .setNeedsAttributes(true) here?

newly added method doesn't include `setNeedsAttributes`, it's a `setNeedsOrderingKey`:

```
public static Read<PubsubMessage> readMessagesWithAllAttributesAndMessageIdAndOrderingKey() 
{
    return Read.newBuilder()
     .setCoder(PubsubMessageCoder.of())
     .setNeedsOrderingKey(true)
     .build();
}
```